### PR TITLE
Add integration tests to Travis CI jobs

### PR DESCRIPTION
**Pull request by ilackarms**
This PR adds Integration tests will attempt to pull and scan an image without throwing an error. In order to accomplish this with our CI (travis), travis now runs tests inside a docker container (as travis does not support centos-based builds).

The PR also adds a set of integration tests for the WebDAV server (ensures http endpoints behave as expected).

the success of `docker build` will now also be tested by travis.
 
Additional changes:
- changes to some constant names
- `ImageServer` interface now implements `GetHandler` which returns an `http.Handler` rather than `ServeImage`, allowing the image server to be tested, as well as for middleware to be more easily added to the http Handler chain.

The additional tests are implemented with the [Ginkgo Test Framework](https://github.com/onsi/ginkgo). Ginkgo tests are compatible with `go test`

### CODE REVIEW

**Comment by ilackarms** [comment]
requesting review from @pweil- @mfojtik

**Comment by ilackarms** [comment]
@enoodle help with review? I think it's important we merge this to help prevent regressions & provide confidence when making changes

File: pkg/inspector/image-inspector.go [hunk 1]
```diff
@@ -154,15 +154,17 @@ func (i *defaultImageInspector) Inspect() error {
...
-		}
 
-		report := reportObj.(openscap.OpenSCAPReport)
-		scanReport = report.ArfBytes
-		htmlScanReport = report.HTMLBytes
+			report := reportObj.(openscap.OpenSCAPReport)
+			scanReport = report.ArfBytes
+			htmlScanReport = report.HTMLBytes
+
+			scanResults.Results = append(scanResults.Results, results...)
```
Comment (enoodle) [inline pkg/inspector/image-inspector.go]: https://github.com/openshift/image-inspector/pull/60

File: pkg/inspector/image-inspector.go [hunk 2]
```diff
@@ -185,7 +187,12 @@ func (i *defaultImageInspector) Inspect() error {
 	}
 
 	if i.imageServer != nil {
-		return i.imageServer.ServeImage(&i.meta, scanResults, scanReport, htmlScanReport)
+		handler, err := i.imageServer.GetHandler(&i.meta, scanResults, scanReport, htmlScanReport)
+		if err != nil {
+			return fmt.Errorf("failed to initialize imageserver: %v", err)
+		}
+		log.Printf("Serving image content %s on webdav://%s%s", i.opts.DstPath, i.opts.Serve, CONTENT_URL_PREFIX)
+		return http.ListenAndServe(i.opts.Serve, handler)
```
Comment (ilackarms) [inline pkg/inspector/image-inspector.go]: unfortunately that makes testing Image server a lot harder (the idiomatic way to test http endpoints is using `httptest` which requires an `http.Handler`)

File: pkg/inspector/image-inspector.go (hunk 1 above)
Comment (ilackarms) [inline pkg/inspector/image-inspector.go]: yep; when that gets merged i'll rebase this. i can remove it from the committed changes here until then, but it's annoying to get panics when trying to test error cases

File: pkg/inspector/image-inspector.go (hunk 2 above)
Comment (enoodle) [inline pkg/inspector/image-inspector.go]: You can add a Serve function that will use the new `GetHandler`. This will be both easy to test and easy to use.

Comment (ilackarms) [inline pkg/inspector/image-inspector.go]: :+1: sounds good

**Comment by ilackarms** [comment]
@enoodle done :) although i think the test should be renamed from "e2e" since they are more like unit tests for the CLI

File: .travis.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    cat > Dockerfile.travis <<EOF
+FROM travis/image-inspector-base
+RUN yum install -y \
+    git \
+    which \
+    make
+RUN yum remove -y golang
+RUN curl -O https://storage.googleapis.com/golang/go1.8.3.linux-amd64.tar.gz && \
+    tar -C /usr/local -xzf go1.8.3.linux-amd64.tar.gz && \
+    rm -f  go1.8.3.linux-amd64.tar.gz
```
Comment (simon3z) [inline .travis.sh]: @ilackarms let's use `openshift/origin-release:golang-1.7` instead.

File: .travis.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+function build_container(){
+    docker build -t travis/image-inspector-base .
+    cat > Dockerfile.travis <<EOF
```
Comment (simon3z) [inline .travis.sh]: @ilackarms I think we mentioned to split this to another file. Have I missed a reply on why this belongs here instead?

File: pkg/imageserver/webdav.go [hunk 5]
```diff
@@ -13,31 +13,30 @@ import (
...
+	chrootServePath = "/"
+	// authTokenHeader is the custom HTTP Header used
 	// to authenticate to image inspector.
-	// We use a custom auth header instead of Authorization
+	// Clients must use a custom auth header
+	// instead of standard Authorization
 	// because Kubernetes Proxy strips the default Auth Header
 	// from requests
-	AUTH_TOKEN_HEADER = "X-Auth-Token"
+	authTokenHeader = "X-Auth-Token"
```
Comment (simon3z) [inline pkg/imageserver/webdav.go]: @ilackarms these are good changes, if you split them out to another PR will be merged quickly I suppose.

Let's not mix refactoring and new features.

**Comment by simon3z** [comment]
@ilackarms if you can split out refactoring (e.g. variable names, `mux.HandleFunc`, etc.) I think it will be easier to review/merge.

File: .travis.sh (hunk 4 above)
Comment (ilackarms) [inline .travis.sh]: Nope; I'll update now

File: .travis.sh (hunk 3 above)
Comment (ilackarms) [inline .travis.sh]: I'll need to change one of the tests because `reflect.DeepEquals` was changed in go 1.8 (making it much easier to compare objects that share a type). I was waiting for the merge of #59 but I can go ahead and make that change now

**Comment by ilackarms** [comment]
I can split this PR into two; the original intent was to a bit of code cleanup. I'll keep this PR focused on testing, and move the refactoring to another branch

**Comment by ilackarms** [comment]
@simon3z i'd like to rebase this branch off of #64 instead of completely removing the refactor changes; they're necessary for `webdav_test.go`

is that alright?

**Comment by ilackarms** [comment]
closing this PR; see #64 (refactoring) and #65 (testing)
